src/storage/heap_file.c

```diff
@@ -25141,6 +25192,17 @@ heap_get_visible_version_internal (THREAD_ENTRY * thread_p, HEAP_GET_CONTEXT * c
       /* we need class_oid to check if the class is mvcc enabled */
       context->class_oid_p = &class_oid_local;
     }
+  // *INDENT-OFF*
+  scope_exit <std::function<void (void)>> reset_class_out_ftor (
```

Comment:
change not related to the main PR; it seemed like a potential bug

Reply:
You might be right.
I also checked and there is no apparent link between `is_heap_scan` and the fact that the home page watcher is saved in `scan_cache`. The save in `scan_cache` happens in `heap_get_visible_version_internal` -> `heap_prepare_get_context` -> `heap_clean_get_context`